data-processing-lib/python/src/data_processing/data_access/data_access_s3.py

```diff
@@ -51,19 +51,17 @@ def __init__(
         self.s3_credentials = {} | s3_credentials
         access_key = self.get_access_key()
         if access_key is None:
+            logger.warning("S3 access key not provided. Defaulting to ''")
             access_key = ""
         secret_key = self.get_secret_key()
         if secret_key is None:
+            logger.warning("S3 secret key not provided. Defaulting to ''")
```

Comment:
I do not see much point in this. If the secret key is not provided, we should exit (throw an exception). Data Access factory do just this in validation

Reply:
done

Reply:
Is the data access factory currently checking the `~/.aws/credentials` file when running locally?